runtime-decompiler/src/main/java/org/jrd/backend/communication/FsAgent.java

```diff
@@ -46,17 +47,18 @@ public FsAgent(List<File> cp) {
     public String submitRequest(final String request) {
         String[] q = request.split("\\s+");
         try {
-            switch (q[0]) {
-                case "CLASSES":
+            switch (AgentRequestAction.RequestAction.fromString(q[0])) {
+                case CLASSES:
                     return readClasses();
-                case "BYTES":
+                case BYTES:
                     return sendByteCode(request);
-                case "OVERWRITE":
+                case OVERWRITE:
                     uploadByteCode(request);
                     return "OK";
-                case "INIT_CLASS":
-                    throw new RuntimeException("Init class have no meaning in FS 'vm'");
-                case "HALT":
+                case INIT_CLASS:
+                    Logger.getLogger().log(Logger.Level.DEBUG, "Init class have no meaning in FS 'vm'");
```

Comment:
Is this reachable from the GUI? If so, why not also log it into your brand new GlobalConsole?

Reply:
yes it is

Reply:
it logs to the brand new GlobalConsole from looger. See the conenction from logger to  brand new globalconsole